pkg/registry/thirdpartyresourcedata/codec.go

```diff
@@ -328,6 +427,8 @@ func (t *thirdPartyResourceDataCodec) EncodeToStream(obj runtime.Object, stream
 		return nil
 	case *unversioned.Status:
 		return t.delegate.EncodeToStream(obj, stream, overrides...)
+	case *unversioned.APIResourceList:
```

Comment:
The trouble here is that this codec gets involved in the API serialize/deserialize parts of discovery so you need to be able to handle that object here, but if it is called for any other kind of real system object, that should be an error.